src/main/java/net/dv8tion/jda/internal/entities/GuildImpl.java

```diff
@@ -1436,7 +1436,7 @@ public long getIdLong()
         {
             if (voiceState.getChannel() == null)
                 throw new IllegalStateException("Can only mute members who are currently in a voice channel");
-            if (voiceState.isGuildMuted() == mute)
+            if (voiceState.isGuildMuted() == mute && !voiceState.isSuppressed())
```

Comment:
I'm unsure if this check makes sense for `mute == true` vs `mute == false`

Reply:
Since the `isGuildMuted` value does not represent whether the member can actually talk (the member cannot talk if he is not muted but suppressed) we have to ignore the local voice state and always tell the api to unmute the member. Telling the API `mute: false` does not only change the guild muted state, but also un-suppresses the member.

Reply:
My point is that the `mute` parameter can be true, in which case the suppressed state is irrelevant.

Reply:
If `mute` is true, then the suppressed state shouldn't be checked at all. Since you don't unsuppress or suppress this way anyway.